### PR TITLE
PublicDashboards: Add react-hook-form for Public Dashboard modal

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

react-hook-form added in Public Dashboard modal

**Why do we need this feature?**

The form is growing and we can manage it in a better way with this library

**Which issue(s) does this PR fix?**:

Fixes [#511](https://github.com/grafana/grafana-enterprise/issues/4587)

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx [hunk 1]
```diff
@@ -45,6 +45,17 @@ import { ShareModal } from '../ShareModal';
...
+type SharePublicDashboardAcknowledgmentInputs = {
+  publicAcknowledgment: boolean;
+  dataSourcesAcknowledgment: boolean;
+  usageAcknowledgment: boolean;
+};
+
+export type SharePublicDashboardInputs = {
+  isAnnotationsEnabled: boolean;
+  enabledSwitch: boolean;
+} & SharePublicDashboardAcknowledgmentInputs;
```
Comment (owensmallwood) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx:57]: whats the `& SharePublicDashboardAcknowledgmentInputs` do?

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx [hunk 2]
```diff
@@ -88,21 +104,21 @@ export const SharePublicDashboard = (props: Props) => {
...
+      isAnnotationsEnabled: publicDashboard?.annotationsEnabled,
+      enabledSwitch: publicDashboard?.isEnabled,
+    });
+  }, [publicDashboard, reset]);
 
   const isLoading = isGetLoading || isSaveLoading || isUpdateLoading;
   const hasWritePermissions = contextSrv.hasAccess(AccessControlAction.DashboardsPublicWrite, isOrgAdmin());
-  const acknowledged = acknowledgements.public && acknowledgements.datasources && acknowledgements.usage;
+  const acknowledged =
+    watch('publicAcknowledgment') && watch('dataSourcesAcknowledgment') && watch('usageAcknowledgment');
```
Comment (owensmallwood) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx:120]: `watch()` gets the current form value?

**Review by owensmallwood — APPROVED** [review]
lgtm 👍

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx (hunk 1 above)
Comment (owensmallwood) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx:57]: neat, learned something new. I knew there were union types, guess it makes sense that the opposite exists

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx (hunk 2 above)
Comment (juanicabanas) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx:120]: correct! anytime the value changes, I can get that value with that method

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx (hunk 1 above)
Comment (juanicabanas) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx:57]: that's correct

**Review by jalevin — APPROVED** [review]
Nice work Juani! Love to see a good refactor.

**Pull request merged**